### PR TITLE
feat: change username and password capabilities

### PR DESCRIPTION
**Pull request by paulin-mipt**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -37,8 +37,7 @@
         "linux/dist/**": true,
     },
     "editor.codeActionsOnSave": {
-        // "source.organizeImports": true,
-        "source.fixAll.eslint": true,
+        "source.fixAll.eslint": "explicit"
```
Comment (alexeyqu) [inline .vscode/settings.json:40]: rm please

File: server/api/auth.go [hunk 2]
```diff
@@ -423,6 +425,158 @@ func (a *API) handleChangePassword(w http.ResponseWriter, r *http.Request) {
 	auditRec.Success()
 }
 
+func (a *API) handleChangeEmail(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /users/{userID}/changeemail changePassword
```
Comment (alexeyqu) [inline server/api/auth.go]: changeEmail

File: server/api/auth.go [hunk 3]
```diff
@@ -423,6 +425,158 @@ func (a *API) handleChangePassword(w http.ResponseWriter, r *http.Request) {
...
+
+	if err = requestData.IsValid(); err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "changeEmail", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelAuth, auditRec)
+
+	if err = a.app.ChangeEmail(userID, requestData.Password, requestData.NewEmail); err != nil {
```
Comment (alexeyqu) [inline server/api/auth.go:495]: requestData.Password

File: server/app/auth.go [hunk 4]
```diff
@@ -281,3 +281,57 @@ func (a *App) ChangePassword(userID, oldPassword, newPassword string) error {
...
+		return errors.New("invalid username or password")
+	}
+
+	err := a.store.UpdateUserUsernameByID(userID, newUsername)
+	if err != nil {
+		return errors.Wrap(err, "unable to update username")
+	}
+
+	return nil
+}
```
Comment (alexeyqu) [inline server/app/auth.go]: EOF

File: webapp/src/mutator.ts [hunk 5]
```diff
@@ -1196,14 +1196,14 @@ class Mutator {
...
 
-    async importFullArchiveTrello(file: File, signupToken: string): Promise<Record<string, string>[]> {
-        const fileContents = await new Response(file.stream()).text();
+    async importFullArchiveTrello(file: File, signupToken: string): Promise<Array<Record<string, string>>> {
+        const fileContents = await new Response(file.stream()).text()
         const input = JSON.parse(fileContents) as Trello
 
         const memberIdMap = new Map<string, string>()
-        const newUserInfo: Record<string, string>[] = []
+        const newUserInfo: Array<Record<string, string>> = []
```
Comment (alexeyqu) [inline webapp/src/mutator.ts:1204]: not sure why but ok

**Review by alexeyqu — APPROVED** [review]
minor nits but LGTM!

File: webapp/src/mutator.ts (hunk 5 above)
Comment (paulin-mipt) [reply webapp/src/mutator.ts:1204]: eslint typescript plugin (?) did it for me on save
agreed to trust it

**Pull request merged**